rules/sql-server-io-pressure/rule.md

```diff
@@ -0,0 +1,138 @@
+                    WHEN (io_stall / (num_of_reads + num_of_writes)) BETWEEN 16 AND 100 THEN 'Poor' 
+                    WHEN (io_stall / (num_of_reads + num_of_writes)) BETWEEN 100 AND 500 THEN  'Bad' 
+                    ELSE 'Deplorable' END  END, 
+         [Avg KBs/Transfer] =  CASE WHEN (num_of_reads = 0 AND num_of_writes = 0) THEN 0 
+                    ELSE ((([num_of_bytes_read] + [num_of_bytes_written]) / (num_of_reads + num_of_writes)) / 1024) END, 
+         LEFT (mf.physical_name, 2) AS Volume, 
+         LEFT(DB_NAME (vfs.database_id),32) AS [Database Name]
+       FROM sys.dm_io_virtual_file_stats (NULL,NULL) AS vfs  
+       JOIN sys.master_files AS mf ON vfs.database_id = mf.database_id 
+         AND vfs.file_id = mf.file_id 
+       ORDER BY AvgLatency DESC
+```
+
+This then gives somewhere to investigate.
+
+## What the IO related wait types mean
+
+You can find out what types of IO waits are occuring in SQL Server with the following query.
+
+```sql
+
+SELECT r.session_id, r.wait_type, r.wait_time as wait_time_ms
+                                       FROM sys.dm_exec_requests r JOIN sys.dm_exec_sessions s
+                                        ON r.session_id = s.session_id
+                                       WHERE wait_type in (
+                                        'PAGEIOLATCH_EX', 
+                                        'PAGEIOLATCH_SH', 
+                                        'PAGEIOLATCH_UP',
+                                        'WRITELOG',
+                                        'ASYNC_IO_COMPLETION',
+                                        'IO_COMPLETION',
+                                        'BACKUPIO')
+                                       AND is_user_process = 1
+```
+
+If these waits exceed 10-15 milliseconds consistently, I/O is considered a bottleneck.
+
+### PAGEIOLATCH_EX
+
+Occurs when a task is waiting on a latch for a data or index page (buffer) in an I/O request. The latch request is in the Exclusive mode. An Exclusive mode is used when the buffer is being written to disk. Long waits may indicate problems with the disk subsystem.
+
+### PAGEIOLATCH_SH
+
+Occurs when a task is waiting on a latch for a data or index page (buffer) in an I/O request. The latch request is in the Shared mode. The Shared mode is used when the buffer is being read from the disk. Long waits may indicate problems with the disk subsystem.
+
+### PAGEIOLATCH_UP
+
+Occurs when a task is waiting on a latch for a buffer in an I/O request. The latch request is in the Update mode. Long waits may indicate problems with the disk subsystem.
+
+### WRITELOG
+
+Occurs when a task is waiting for a transaction log flush to complete. A flush occurs when the Log Manager writes its temporary contents to disk. Common operations that cause log flushes are transaction commits and checkpoints.
+
+Common reasons for long waits on WRITELOG are:
```

Comment:
**issue**: The paragraphs that follow should be dot-points.

Reply:
Turned them into level 5 headings, they were horrible as dot points :)